modules/app/hooks/useAccount.ts

```diff
@@ -0,0 +1,69 @@
+import { ethers } from 'ethers';
+import { useCurrentUserVoteDelegateContract } from 'modules/delegates/hooks/useCurrentUserVoteDelegateContract';
+import { useVoteDelegateAddress } from 'modules/delegates/hooks/useVoteDelegateAddress';
+import { useActiveWeb3React } from 'modules/web3/hooks/useActiveWeb3React';
+import { useCurrentUserVoteProxyContract } from './useCurrentUserVoteProxyContract';
+import { useCurrentUserVoteProxyOldContract } from './useCurrentUserVoteProxyOldContract';
+import { useVoteProxyAddress } from './useVoteProxyAddress';
+import { useVoteProxyOldAddress } from './useVoteProxyOldAddress';
+
+type UseAccountResponse = {
+  account?: string | null;
+
+  voteDelegateContract?: ethers.Contract;
+  voteDelegateContractAddress?: string;
+
+  voteProxyContract?: ethers.Contract;
+  voteProxyContractAddress?: string;
+  voteProxyHotAddress?: string;
+  voteProxyColdAddress?: string;
+
+  voteProxyOldContract?: ethers.Contract;
+  voteProxyOldContractAddress?: string;
+  voteProxyOldHotAddress?: string;
+  voteProxyOldColdAddress?: string;
+
+  mutate: () => void;
+};
+
+export function useAccount(): UseAccountResponse {
+  const { account } = useActiveWeb3React();
+
+  if (!account) {
+    return {
+      account,
+      mutate: () => null
+    };
+  }
```

Comment:
So I think there is a problem with this implementation, we are conditionally rendering the hooks based on `account` which breaks one of the rules of hooks. If we don't have an account, we don't run the hooks below, but when we finally get it, we do. React doesn't like that.

I think we need to make the hooks handle the case of `!account` so we can run each of the hooks in the same order every time. One way to do this is with useSWR which does allow for [conditional fetching](https://swr.vercel.app/docs/conditional-fetching), eg: `const { data, error } = useSWR(contractAddress ? '${contractAddress}/vote-delegate-contract' : null, ...` or put the actual contract calls in a non-hook and return that conditionally, eg:`getChiefDeposits` here: 
```
const { data, error, mutate } = useSWR(
    addressToCache ? '${chief.address}/user/mkr-locked${addressToCache}' : null,
    async () => {
      return addressToCache ? await getChiefDeposits(addressToCache, chief) : null;
    }
  );
  ```

Reply:
fixed